File: gui/src/ts/models/benchmark/BenchmarkQuestion.ts
import {
  BenchmarkQuestionPromptType,
  BenchmarkQuestionType,
} from "api/graphqlGlobalTypes";

export interface BenchmarkQuestion {
  answers?: {
    choiceId: string;
    submissionId: string;
  }[];
  categories: string[] | null;
  concept: string | null;
  correctChoiceId: string;
  id: string;
  prompts: {
    text: string;
    type: BenchmarkQuestionPromptType;
  }[];
  type: BenchmarkQuestionType | null;
}

File: lib/scala/benchmark/src/main/scala/io/github/tetherlessworld/mcsapps/lib/benchmark/models/BenchmarkQuestion.scala
package io.github.tetherlessworld.mcsapps.lib.benchmark.models

final case class BenchmarkQuestion(
                                    categories: Option[List[String]],
                                    choices: List[BenchmarkQuestionChoice],
                                    concept: Option[String],
                                    correctChoiceId: String,
                                    id: String,
                                    datasetId: String,
                                    prompts: List[BenchmarkQuestionPrompt],
                                    `type`: Option[BenchmarkQuestionType]
                                  )
